src/transaction_builder.cpp

```diff
@@ -128,15 +128,15 @@ boost::optional<CTransaction> TransactionBuilder::Build()
         // Send change to the specified change address. If no change address
         // was set, send change to the first Sapling address given as input.
         if (zChangeAddr) {
-            AddSaplingOutput(zChangeAddr->first, zChangeAddr->second, change, {});
+            AddSaplingOutput(zChangeAddr->first, zChangeAddr->second, change);
         } else if (tChangeAddr) {
             // tChangeAddr has already been validated.
             assert(AddTransparentOutput(tChangeAddr.value(), change));
         } else if (!spends.empty()) {
             auto fvk = spends[0].expsk.full_viewing_key();
             auto note = spends[0].note;
             libzcash::SaplingPaymentAddress changeAddr(note.d, note.pk_d);
-            AddSaplingOutput(fvk, changeAddr, change, {});
```

Comment:
Is it incorrect to pass in an empty vector? I ask because it is still possible to do this, for example, we do this in `test_transaction_builder.cpp` and `test_wallet.cpp`. I was wondering if it was something that should be enforced in the `TransactionBuilder::Build()` method.

Reply:
Yeah, this is something I messed up in the initial implementation. We denote an empty memo field with an invalid UTF-8 byte, and then all-zeros afterwards. If the leading byte is a valid UTF-8 character (of which 0 is), then the entire field is to be treated as a UTF-8 string, rather than empty.

Reply:
It's not something to enforce in `Build()`, because if the user *wants* the field to be the all-zeroes UTF-8 array, then fine. It's about being consistent as to how we signal "there is no memo field here".